vernac/lemmas.ml

```diff
@@ -359,9 +359,9 @@ let start_proof id ?pl kind sigma ?terminator ?sign c ?init_tac ?(compute_guard=
     | None -> initialize_named_context_for_proof ()
   in
   !start_hook c;
-  Pfedit.start_proof id ?pl kind sigma sign c ?init_tac terminator
+  Pfedit.start_proof id ~allow_nested:true ?pl kind sigma sign c ?init_tac terminator
```

Comment:
I could also add the `~allow_nested` parameter to `Lemmas.start_proof` to show further where it's being called (it is called in Funind in particular).

Reply:
For instance the call in classes.ml should respect the option.

Reply:
OK

Reply:
Yup there are quite a few callers, so maybe you want instead of plug the option at the STM level? In fact `pstates` may be close to be gone, you can check nesting quite easily in `process_transaction` I think.